modules/RTC/ortc/RTCPeerConnection.js

```diff
@@ -0,0 +1,532 @@
+        if (this._closed) {
+            return Promise.reject(
+                new InvalidStateError('RTCPeerConnection closed'));
+        }
+
+        switch (desc.type) {
+        case 'offer':
+            if (this.signalingState !== 'stable') {
+                return Promise.reject(new InvalidStateError(
+                    `invalid signalingState "${this.signalingState}"`));
+            }
+
+            break;
+
+        case 'answer':
+            if (this.signalingState !== 'have-remote-offer') {
+                return Promise.reject(new InvalidStateError(
+                    `invalid signalingState "${this.signalingState}"`));
+            }
+
+            break;
+        }
+
+        // TODO: More stuff.
+    }
+
+    /**
+     */
+    _setRemoteDescription(desc) {
+        if (this._closed) {
+            return Promise.reject(
+                new InvalidStateError('RTCPeerConnection closed'));
+        }
+
+        switch (desc.type) {
+        case 'offer':
+            if (this.signalingState !== 'stable') {
+                return Promise.reject(new InvalidStateError(
+                    `invalid signalingState "${this.signalingState}"`));
+            }
+
+            break;
+
+        case 'answer':
+            if (this.signalingState !== 'have-local-offer') {
+                return Promise.reject(new InvalidStateError(
+                    `invalid signalingState "${this.signalingState}"`));
+            }
+
+            break;
+        }
+
+        // TODO: More stuff.
+    }
+
+    /**
+     */
+    _addStream(stream) {
+        if (this._closed) {
+            throw new InvalidStateError('RTCPeerConnection closed');
+        }
+
+        // TODO: Not sure. The app may wish to signal that the same stream
+        // has now new or removed tracks.
+        if (this._localStreams.has(stream)) {
+            return;
+        }
+
+        this._localStreams.add(stream);
+
+        // It may need to renegotiate.
+        this._emitNegotiationNeeded();
+    }
+
+    /**
+     */
+    _removeStream(stream) {
+        if (this._closed) {
+            throw new InvalidStateError('RTCPeerConnection closed');
+        }
+
+        if (!this._localStreams.has(stream)) {
+            return;
+        }
+
+        this._localStreams.delete(stream);
+
+        // It may need to renegotiate.
+        this._emitNegotiationNeeded();
+    }
+
+    /**
+     */
+    _updateAndEmitSignalingStateChange(state) {
+        if (state === this.signalingState) {
+            return;
+        }
+
+        this._signalingState = state;
+
+        logger.debug(
+            'emitting "signalingstatechange", signalingState:',
+            this.signalingState);
+
+        const event = new yaeti.Event('signalingstatechange');
+
+        this.dispatchEvent(event);
+    }
+
+    /**
+     */
+    _emitNegotiationNeeded() {
+        // Ignore if signalingState is not 'stable'.
+        if (this.signalingState !== 'stable') {
```

Comment:
Why? I did use constants at the beginning in a similar code I removed them due bad readability. Do you really think it's better to have the following?:

```js
if (this.signalingState !== RTCSignalingState.stable) {
```